xwiki-pro-macros-ui/src/main/resources/XWiki/Macros/Team.xml

```diff
@@ -247,17 +246,11 @@
     }
   }
 
-  function init() {
+  $(function () {
     initTools();
-    document.observe('xwiki:dom:updated', initTools);
-  }
-
-  if (document.body) {
-    init();
-  } else {
-    window.addEventListener('DOMContentLoaded', init);
-  }
-})();</code>
+    $('xwiki:dom:updated', initTools);
```

Comment:
You're still iterating all the ``xwikiteam`` elements whenever ``xwiki:dom:updated`` is triggered (instead of looking **just** at the updated elements, which you can get from the event data), but you're initializing only the ``xwikiteam`` elements that are not already initialized, which I guess it's fine. My worry is that the ``initialized`` CSS class is not bulletproof. Take this use case:

* you are inside the WYSIWYG editor
* the found ``xwikiteam`` elements are initialized, thus getting the ``initialized`` CSS class
* the editor saves the current HTML (in order to be able to Undo / Redo)
* the user makes some changes and then clicks on Undo
* the editor reverts the HTML and triggers the ``xwiki:dom:updated`` event
* the ``xwikiteam`` elements are **not** re-initialized because they have the ``initialized`` CSS class

I haven't tested but this user case make sense. All in all, I think it's safer to re-initialize everything from the updated content indicated by ``xwiki:dom:updated``.

Reply:
Fixed by removing the use of a class and removing / adding the event unconditionally.